src/routes/create_new_engagement/create_new_engagement.tsx

```diff
@@ -295,6 +335,58 @@ export function CreateNewEngagementForm() {
                   )}
                 </FormSelect>
               </FormGroup>
+              <FormGroup
+                label={<SectionTitle>Copy From&nbsp;&nbsp;</SectionTitle>}
+                fieldId="copyFrom"
+                helperText="Prepopulate details of this engagement based on a selected template or an existing engagement"
+              >
+                <FormSelect
+                  data-testid="new-engagement-copy-from"
+                  data-cy="new-engagement-copy-from"
+                  value={searchTerm}
+                  onChange={e => {
+                    setSearchTerm(e);
+                  }}
+                >
+                  <FormSelectOption
+                    value={undefined}
+                    label="None"
+                    key="undefined engagement"
+                  />
+                  <FormSelectOptionGroup label={'Templates'}>
+                    {
+                      <FormSelectOption
+                        isDisabled={true}
+                        value={undefined}
+                        label="No templates available"
+                        key="undefined template"
+                      />
+                    }
+                  </FormSelectOptionGroup>
+                  <FormSelectOptionGroup label={'Existing engagements'}>
+                    {
+                      engagements?.
+                      sort(
+                        function(a, b){
+                          if(a.customer_name < b.customer_name) { return -1; }
+                          if(a.customer_name > b.customer_name) { return 1; }
+                          return 0;
+                        }
+                      ).
+                      map(
+                        engagement => {
+                          return (
+                            <FormSelectOption
+                              label={engagement.customer_name + ' - ' + engagement.project_name }
+                              key={engagement.project_name}
```

Comment:
Project names are not unique by themselves, they are only unique in conjunction with customer name. This needs to to compare based on the value of both customer and project name